readthedocs/oauth/tasks.py

```diff
@@ -22,20 +31,12 @@
 
 
 @permission_check(user_id_matches)
-class SyncRemoteRepositories(PublicTask):
-
-    name = __name__ + '.sync_remote_repositories'
-    public_name = 'sync_remote_repositories'
-    queue = 'web'
-
-    def run_public(self, user_id):
-        user = User.objects.get(pk=user_id)
-        for service_cls in registry:
-            for service in service_cls.for_user(user):
-                service.sync()
-
-
-sync_remote_repositories = SyncRemoteRepositories()
+@app.task(queue='web', base=PublicTask)
+def sync_remote_repositories(user_id):
+    user = User.objects.get(pk=user_id)
+    for service_cls in registry:
+        for service in service_cls.for_user(user):
+            service.sync()
```

Comment:
The api for defining a public task is cleaner now :D

Reply:
Much nicer!